File: NEETCODE/Easy/binary-search.js
// https://neetcode.io/problems/binary-search

class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let start = 0;
        let end = nums.length - 1;
        while (start <= end) {
            let mid = Math.floor(start + (end - start) / 2)
            if (nums[mid] === target) {
                return mid
            }
            if (nums[mid] > target) {
                end = mid - 1
            } else {
                start = mid + 1
            }
        }
        return -1
    }
}

File: NEETCODE/Easy/binary-search.py
# https://neetcode.io/problems/binary-search

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start, end = 0, len(nums)
        mid = end // 2
        while start < end:
            if nums[mid] == target:
                return mid

            if nums[mid] > target:
                end = mid
            else:
                start = mid + 1
            mid = (start + end) // 2
        return -1